src/main/java/liqp/tags/Include.java

```diff
@@ -14,26 +14,33 @@ public class Include extends Tag {
     public static final String INCLUDES_DIRECTORY_KEY = "liqp@includes_directory";
     public static String DEFAULT_EXTENSION = ".liquid";
 
+    @SuppressWarnings("deprecation")
     @Override
     public Object render(TemplateContext context, LNode... nodes) {
 
         try {
             String includeResource = super.asString(nodes[0].render(context), context);
             String extension = DEFAULT_EXTENSION;
-            if(includeResource.indexOf('.') > 0) {
+            if (includeResource.indexOf('.') > 0) {
                 extension = "";
             }
             File includeResourceFile;
             String includesDirectory = (String) context.get(INCLUDES_DIRECTORY_KEY);
 
             if (includesDirectory != null) {
                 includeResourceFile = new File(includesDirectory, includeResource + extension);
-            }
-            else {
-                includeResourceFile = new File(context.parseSettings.flavor.snippetsFolderName, includeResource + extension);
+            } else {
+                includeResourceFile = new File(context.parseSettings.flavor.snippetsFolderName,
+                        includeResource + extension);
             }
 
-            Template template = Template.parse(includeResourceFile, context.parseSettings, context.renderSettings);
+            Template template;
+            if (context.getParser().isLegacyMode()) {
+                template = Template.parse(includeResourceFile, context.getParseSettings(), context
+                        .getRenderSettings());
+            } else {
+                template = context.getParser().parse(includeResourceFile);
+            }
 
```

Comment:
Good point. Actually, I think that was a bug here. In the case of a custom templator this is expected that includes will be parsed with it (but not by the one from a static context). Not sure we even need this flag as we can simply go new approach exclusively.